src/main/java/com/mgmtp/radio/service/station/StationServiceImpl.java

```diff
@@ -1,23 +1,84 @@
 package com.mgmtp.radio.service.station;
 
+import com.mgmtp.radio.domain.station.SkipRule;
 import com.mgmtp.radio.domain.station.Station;
-import com.mgmtp.radio.respository.station.StationRepository;
-import org.springframework.stereotype.Service;
-import reactor.core.publisher.Mono;
+import com.mgmtp.radio.domain.station.StationConfiguration;
+import com.mgmtp.radio.dto.station.SongDTO;
+import com.mgmtp.radio.dto.station.StationConfigurationDTO;
 import com.mgmtp.radio.dto.station.StationDTO;
+import com.mgmtp.radio.dto.user.UserDTO;
 import com.mgmtp.radio.mapper.station.StationMapper;
+//import com.mgmtp.radio.mapper.stationConfiguration.StationConfigurationMapper;
+//import com.mgmtp.radio.respository.station.StationConfigurationRepository;
+import com.mgmtp.radio.respository.station.StationRepository;
+import org.aspectj.lang.annotation.AfterReturning;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.stereotype.Service;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
 import java.time.LocalDate;
 import java.util.List;
 
-@Service
+@Service("stationService")
+@Aspect
 public class StationServiceImpl implements StationService {
 
-    private final StationMapper stationMapper;
+	private final StationMapper stationMapper;
+	private static final double DOWN_VOTE_THRES_PERCENT = 0.5;
+
+	@Override
+	public int getOnlineUsersNumber(StationDTO stationDTO) {
+		//TODO Get number of online users id here
+		return 0;
+	}
+
+	private double calcCurrentSongDislikePercent(SongDTO songDTO, StationDTO station) {
+		final int numberOnline = getOnlineUsersNumber(station);
+		double currentSongDislikePercent = 0;
+		if (numberOnline > 0) {
+			currentSongDislikePercent = songDTO.getDownVoteCount() / (float) numberOnline;
+		}
+		return currentSongDislikePercent;
+	}
+
+	private boolean isOwnerDownvote(Station station, SongDTO songDTO) {
+		for (UserDTO user: songDTO.getDownvoteUserList()) {
+			if (station.getOwnerId().equals(user.getId())) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@AfterReturning(value = "execution(* com.mgmtp.radio.service.station.SongService.downVoteSongInStationPlaylist(..))", returning = "monoSongDTO")
+	public Mono<SongDTO> checkAndSkipSongIfNeeded(Mono<SongDTO> monoSongDTO) {
+		Mono<SongDTO> ms = monoSongDTO.map(songDTO -> {
```

Comment:
what is ms mean? Why dont you return here?

Reply:
what do you mean, @dat-ng ?  " Why don't you return here?"